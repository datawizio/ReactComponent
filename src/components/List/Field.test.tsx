import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import ListItemField from "./Field";

const mockProps = {
  title: "title",
  description: "description",
  value: "value",
  onClick: jest.fn()
};

const setUp = (props?) => mount(<ListItemField {...props} />);

describe("ListItemField component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("ListItemField rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("ListItemField correctly call click action", () => {
    const item = component.find(".list-item-field");
    item.first().simulate("click");
    expect(component.props().onClick).toBeCalledTimes(1);
  });

  it("ListItemField render  with correct value", () => {
    const item = component.find(".value-container");
    expect(item.html()).toContain(mockProps.value);
  });
});
