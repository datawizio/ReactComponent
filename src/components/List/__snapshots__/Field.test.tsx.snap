// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemField component ListItemField rendered correctly 1`] = `
<ListItemField
  description="description"
  onClick={[MockFunction]}
  title="title"
  value="value"
>
  <Item
    className="list-item-field"
    onClick={[MockFunction]}
  >
    <Item
      className="list-item-field"
      onClick={[MockFunction]}
    >
      <li
        className="ant-list-item list-item-field"
        onClick={[MockFunction]}
      >
        <div
          className="list-item-field-container"
        >
          <Meta
            description="description"
            title="title"
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  title
                </h4>
                <div
                  className="ant-list-item-meta-description"
                >
                  description
                </div>
              </div>
            </div>
          </Meta>
          <div
            className="value-container"
          >
            value
          </div>
        </div>
        <ForwardRef(RightOutlined)>
          <AntdIcon
            icon={
              Object {
                "icon": Object {
                  "attrs": Object {
                    "focusable": "false",
                    "viewBox": "64 64 896 896",
                  },
                  "children": Array [
                    Object {
                      "attrs": Object {
                        "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                      },
                      "tag": "path",
                    },
                  ],
                  "tag": "svg",
                },
                "name": "right",
                "theme": "outlined",
              }
            }
          >
            <span
              aria-label="right"
              className="anticon anticon-right"
              role="img"
            >
              <IconReact
                icon={
                  Object {
                    "icon": Object {
                      "attrs": Object {
                        "focusable": "false",
                        "viewBox": "64 64 896 896",
                      },
                      "children": Array [
                        Object {
                          "attrs": Object {
                            "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                          },
                          "tag": "path",
                        },
                      ],
                      "tag": "svg",
                    },
                    "name": "right",
                    "theme": "outlined",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  key="svg-right"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                    key="svg-right-svg-0"
                  />
                </svg>
              </IconReact>
            </span>
          </AntdIcon>
        </ForwardRef(RightOutlined)>
      </li>
    </Item>
  </Item>
</ListItemField>
`;
